lib/index.js

```diff
@@ -16,13 +16,23 @@ function wrapError(callback) {
 }
 
 function putUser(db, user, opts, callback) {
-  var reservedWords = ['name', 'password', 'roles', 'type', 'salt', 'metadata'];
+  var reservedWords = [
+    '_id',
+    '_rev',
+    'name',
+    'password',
+    'roles',
+    'type',
+    'salt',
+    'derived_key',
+    'password_scheme',
+    'iterations'
+  ];
+
   if (opts.metadata) {
     for (var key in opts.metadata) {
-      if (opts.hasOwnProperty(key)) {
-        if (reservedWords.indexOf(key) !== -1 || key.startsWith('_')) {
-          return callback(new AuthError('cannot use reserved word in metadata: "' + key + '"'));
-        }
+      if (reservedWords.indexOf(key) !== -1) {
```

Comment:
The `hasOwnProperty` check is not optional; it's standard for these kind of JS for loops

Reply:
@nolanlawson @tlvince We only have to keep the `hasOwnProperty` test if it makes sense to have an `opts` object that is an instance of a class in a class hierarchy. I guess for options to a function this does never occur.

However, if we choose to keep it, it has to be tested on `opts.metadata` and not on `opts` directly, right ?
Currently, this only check for reserved words on properties that are defined both on `opts` and `opts.metadata`.